src/python/turicreate/toolkits/sound_classifier/_tf_sound_classifier.py

```diff
@@ -22,17 +18,16 @@ class SoundClassifierTensorFlowModel(TensorFlowModel):
     def __init__(self, num_inputs, num_classes, custom_layer_sizes):
         """
         Defines the TensorFlow model, loss, optimisation and accuracy.
-
         """
+        from uuid import uuid1
 
-        _tf.reset_default_graph()
-
+        self.instance_id = str(uuid1())
```

Comment:
this won't work since each time predict/predict_topk is called, a new python model object will be created. So every time you create a new id instead of reusing the previous id.

Reply:
I think a class level static variable is already sufficient to provide a unique uuid for each ml model class.

Reply:
A new python model is not created each time `predict` is called. I *think* that's how many of the other deep learning toolkits work, but that's not how the sound classifier works.

Reply:
I see! I thought SC is written in Cpp. My bad. Will this namespace thing solve it? Do we have to disable all v2 functionalities for this release?

Reply:
Adding this UUID seems invasive to me. In resolving #2726 it seemed to suffice to use `tf.Graph.as_default()`, see #2728 

Could we use this PR just to address problem #1 that we discussed? The consistent disabling of v2 behavior?

Reply:
@nickjong - I agree there is probably a better fix than this PR. It sounds like your basically suggesting option 2 from #2731. I agree that would be better. Although I think there may be some subtle issues with code that is actually using the V2 APIs.

Reply:
Agreed! For 6.0, I think only the SC case is P1. It's wrong that we're clobbering global state in the other toolkits, but only SC is actually broken as a result. Anyway, it looks like we can resolve all of the SC issues with #2741